library/mlops/src/iteration.py

```diff
@@ -88,15 +114,28 @@ def end_iteration(self) -> dict:
         """
 
         self.format_path()
-
-        data = {
-            "user_name": self.user_name,
-            "iteration_name": self.iteration_name,
-            "metrics": self.metrics,
-            "parameters": self.parameters,
-            "path_to_model": self.path_to_model,
-            "model_name": self.model_name
-        }
+        if self.hasDataset:
```

Comment:
and here having only `self.dataset_id` we can define data using:
```
if self.dataset_id:
    dataset = {"id": self.dataset_id}
else:
    dataset = None

data = {
    "user_name": self.user_name,
    "iteration_name": self.iteration_name,
    "metrics": self.metrics,
    "parameters": self.parameters,
    "path_to_model": self.path_to_model,
    "model_name": self.model_name,
    "dataset": dataset
}
```

Reply:
corrected